### PR TITLE
JSUI-2394 Closing facet settings on escape key press

### PR DESCRIPTION
**Pull request by samisayegh**
I started on the jira by adding `withKeyboardAction` to the AccessibleButton class. I ultimately did not use it because I needed to add the escape event to the SettingsButton as well as the SettingsPopup which is not a button.

I left the logic in the PR to see if you find it useful, but I am fine rolling back the commit with the changes to the AccessibleButton class b803b8d.

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,7 +146,7 @@
     "ts-loader": "^2.0.3",
     "tvm": "^0.8.14",
     "typedoc": "https://github.com/coveord/typedoc",
-    "typescript": "2.8.1",
+    "typescript": "2.9.2",
```
Comment (olamothe) [inline package.json]: 🤔 ?

Updating is fine, but please do so in a separate story if you need to.

File: src/utils/AccessibleButton.ts [hunk 2]
```diff
@@ -63,6 +64,11 @@ export class AccessibleButton {
     return this;
   }
 
+  public withKeyboardAction(key: KEYBOARD, action: (e: Event) => void) {
```
Comment (olamothe) [inline src/utils/AccessibleButton.ts]: Since it's not used anywhere in the codebase for now, I would remove it.

File: package.json (hunk 1 above)
Comment (olamothe) [reply package.json]: It also seems it breaks the file definition validation on older version of typescript (1.8)

Comment (samisayegh) [reply package.json]: I did not end up needing to update. I reverted this change, thanks.

File: src/utils/AccessibleButton.ts (hunk 2 above)
Comment (samisayegh) [reply src/utils/AccessibleButton.ts]: I reverted this change, thanks.

**Pull request merged**